network/ldap_entry.py

```diff
@@ -0,0 +1,305 @@
+    non-existence of an LDAP entry, not its attributes. To assert the
+    attribute values of an entry, see M(ldap_attr).
+notes:
+  - The default authentication settings will attempt to use a SASL EXTERNAL
+    bind over a UNIX domain socket. This works well with the default Ubuntu
+    install for example, which includes a cn=peercred,cn=external,cn=auth ACL
+    rule allowing root to modify the server configuration. If you need to use
+    a simple bind to access your server, pass the credentials in I(bind_dn)
+    and I(bind_pw).
+version_added: '2.3'
+author:
+  - Jiri Tyr (@jtyr)
+requirements:
+  - python-ldap
+options:
+  '...':
+    required: false
+    default: null
+    description:
+      - If I(state=present), all additional arguments are taken to be
+        LDAP attribute names like I(objectClass), with similar
+        lists of values. These should only be used to
+        provide the minimum attributes necessary for creating an entry;
+        existing entries are never modified. To assert specific attribute
+        values on an existing entry, see M(ldap_attr).
+  bind_dn:
+    required: false
+    default: null
+    description:
+      - A DN to bind with. If this is omitted, we'll try a SASL bind with
+        the EXTERNAL mechanism. If this is blank, we'll use an anonymous
+        bind.
+  bind_pw:
+    required: false
+    default: null
+    description:
+      - The password to use with I(bind_dn).
+  dn:
+    required: true
+    description:
+      - The DN of the entry to add or remove.
+  objectClass:
+    required: false
+    default: null
+    description:
+      - If I(state=present), value or list of values to use when creating
+        the entry. It can either be a string or an actual list of
+        strings.
+  params:
+    required: false
+    default: null
+    description:
+      - Option used to allow the user to overwrite any of the other
+        options. To remove an option, set the value of the option to
+        C(null).
+  server_uri:
+    required: false
+    default: ldapi:///
+    description:
+      - A URI to the LDAP server. The default value lets the underlying
+        LDAP client library look for a UNIX domain socket in its default
+        location.
+  start_tls:
+    required: false
+    choices: ['yes', 'no']
+    default: 'no'
+    description:
+      - If true, we'll use the START_TLS LDAP extension.
+  state:
+    required: false
+    choices: [present, absent]
+    default: present
+    description:
+      - The target state of the entry.
+"""
+
+
+EXAMPLES = """
+- name: Make sure we have a parent entry for users
+  ldap_entry:
+    dn: ou=users,dc=example,dc=com
+    objectClass: organizationalUnit
+
+- name: Make sure we have an admin user
+  ldap_entry:
+    dn: cn=admin,dc=example,dc=com
+    objectClass:
+      - simpleSecurityObject
+      - organizationalRole
+    description: An LDAP administrator
+    userPassword: "{SSHA}tabyipcHzhwESzRaGA7oQ/SDoBZQOGND"
```

Comment:
see my point above, this seems confusing as the 'non existing options' are slurped into the entry definitions, there is no clear separation between the ldap entry properties and the task options. A simple misspelling will also create havok on entries.
```
prams: hello=kitty
```